source/common/http/async_client_utility.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "envoy/http/async_client.h"
+
+namespace Envoy {
+namespace Http {
+
+/**
+ * Keeps track of active async HTTP requests to be able to cancel them on destruction.
+ */
+class AsyncClientRequestTracker {
+public:
+  /**
+   * Cancels all known active async HTTP requests.
+   */
+  ~AsyncClientRequestTracker();
+  /**
+   * Includes a given async HTTP request into a set of known active requests.
+   */
+  AsyncClientRequestTracker& add(AsyncClient::Request* request);
```

Comment:
nit: does this still have to return a reference to itself or can it return void (same below)

Reply:
fixed